From 26a3453079df536b7f85c328380d43201c3fe284 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Sun, 8 Dec 2013 13:47:32 +0000
Subject: [PATCH 064/351] ASoC: wm_adsp: Firmware array should be accessed
 through the ADSP core

A couple of places in the compressed handling code were accessing the
firmware array directly rather than through the ADSP core. This would
cause some problems if the firmware list is specified in device tree.

Change-Id: I3e5e80ec785610f969012e6b3dbeadb4a761db7a
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/wm_adsp.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/sound/soc/codecs/wm_adsp.c b/sound/soc/codecs/wm_adsp.c
index 8a8b526..a9495e9 100644
--- a/sound/soc/codecs/wm_adsp.c
+++ b/sound/soc/codecs/wm_adsp.c
@@ -2433,7 +2433,7 @@ static int wm_adsp_read_buffer(struct wm_adsp *adsp, int32_t read_index,
 	int i, ret;
 
 	/* Calculate read parameters */
-	for (i = 0; i < wm_adsp_fw[adsp->fw].caps->num_host_regions; ++i) {
+	for (i = 0; i < adsp->firmwares[adsp->fw].caps->num_host_regions; ++i) {
 		if (read_index < adsp->host_regions[i].cumulative_size)
 			break;
 	}
@@ -2481,9 +2481,8 @@ static int wm_adsp_read_buffer(struct wm_adsp *adsp, int32_t read_index,
 
 static int wm_adsp_capture_block(struct wm_adsp *adsp, int *avail)
 {
-	int last_region = wm_adsp_fw[adsp->fw].caps->num_host_regions - 1;
-	int host_size =
-		adsp->host_regions[last_region].cumulative_size;
+	int last_region = adsp->firmwares[adsp->fw].caps->num_host_regions -1;
+	int host_size = adsp->host_regions[last_region].cumulative_size;
 	int num_words;
 	u32 next_read_index, next_write_index;
 	int32_t write_index, read_index;
-- 
1.7.9.5

